workflows/execute_workflows.sh

```diff
@@ -84,47 +119,43 @@ do
 
     DIR_NAME=$(echo $WS_DIR | rev | cut -d'/' -f 2 | rev)
 
-    # get all NextFlow files that do not belong to an eval workflow
-#    NF_FILES=("$WS_DIR"/*.nf)
-#    NF_FILES_WO_EVAL=()
-#    EVAL_NFS=()
-#
-#    for NF in "${NF_FILES[@]}"
-#    do
-#        if [[ $NF != *"dinglehopper_eval.txt.nf" ]]; then
-#            NF_FILES_WO_EVAL+=($NF)
-#        else
-#            EVAL_NFS+=($NF)
-#        fi
-#    done
-
-    # … adjust workflow files to corpus and run workflows.
-    for WORKFLOW in "$WS_DIR"/*.nf
+    # separate OCR workflows from evaluation workflows
+    NF_FILES=("$WS_DIR"/*.nf)
+    OCR_NFS=()
+    EVAL_NFS=()
+
+    for NF in "${NF_FILES[@]}"
+    do
+        if [[ $NF != *"dinglehopper_eval.txt.nf" ]]; then
+            OCR_NFS+=($NF)
+        else
+            EVAL_NFS+=($NF)
+        fi
+    done
+
+    # run OCR workflows
+    for WORKFLOW in "${OCR_NFS[@]}"
+    do
+        run "$WORKFLOW" "$DIR_NAME"
+        save_workspaces "$WS_DIR" "$DIR_NAME" "$WORKFLOW"
+    done
+
+    # run eval workflows
+    for WORKFLOW in "${EVAL_NFS[@]}"
     do
-        sed -i "s INPUT_DIR . g" "$WORKFLOW"
-        sed -i "s OUTPUT_DIR result g" "$WORKFLOW"
-        sed -i "s CURRENT $DIR_NAME g" "$WORKFLOW"
-        nextflow run "$WORKFLOW" -with-weblog http://127.0.0.1:8000/nextflow/
-
-        # package workspace
-        echo "Zipping workspace $WS_DIR"
-        ocrd zip bag -d $WS_DIR -i $WS_DIR $WS_DIR
-
-        # create a result JSON according to the specs          
-        echo "Get Benchmark JSON …"
-        VENV=$ROOT'/venv/bin/python'
-        CMD="$ROOT/quiver/benchmark_extraction.py $WS_DIR $WORKFLOW"
-        $VENV $CMD
-        echo "Done."
-        for DATA in "$ROOT"/workflows/nf-results/*
-        do
-            rm -rf "$DATA"
-        done
+        run "$WORKFLOW" "$DIR_NAME"
+        save_workspaces "$WS_DIR" "$DIR_NAME" "$WORKFLOW"
     done
 
+    # create a result JSON according to the specs          
+    echo "Get Benchmark JSON …"
+    VENV=$ROOT'/venv/bin/python'
+    CMD="$ROOT/quiver/benchmark_extraction.py $WS_DIR $WORKFLOW"
```

Comment:
It would be better to properly install the python packages instead of calling scripts directly. Also for oton. These should be installed via `pip` and use the CLI defined in its `entry_points`. The script should activate the venv as a first step (unless `$VIRTUAL_ENV` is already set, ie. venv is already active).

Reply:
I'm not sure I get what you mean. I do install OtoN via `pip` (https://github.com/OCR-D/quiver-back-end/pull/42/files#diff-f3a5af4cc255c1a06e9fa9c6732102396989b318b4ac2a0fd52535a41d525793R77) and use its CLI (https://github.com/OCR-D/quiver-back-end/pull/42/files#diff-f3a5af4cc255c1a06e9fa9c6732102396989b318b4ac2a0fd52535a41d525793R91).

Since OtoN is not on PyPi, so I saw no other way to install it. But I'd be glad to learn a better way :)

(Maybe we should ask Mehmed to put OtoN on PyPi, though.)